File: georgia_tech/sgl-eval-peterhj-genserver/website/src/api_types/questions.ts
export type Message = {
  role: "assistant" | "user";
  content: string;
};

export type QuestionAttemptAnswer = {
  index: number;
  score: number;
  messages: Message[];
};

export type QuestionLlmAnswer = {
  llm_id: string;
  llm_name: string;
  attempts: QuestionAttemptAnswer[];
};

export interface QuestionResponse {
  id: string;
  difficulty: number;
  benchmark_category: string;
  question: string;
  llm_answers: QuestionLlmAnswer[];
} 

File: georgia_tech/sgl-eval-peterhj-genserver/backend/eval_next_backend/api_types/questions.py
from typing import List
from pydantic import BaseModel

class Message(BaseModel):
    role: str
    content: str

class QuestionAttemptAnswer(BaseModel):
    index: int
    score: float
    messages: List[Message]

class QuestionLlmAnswer(BaseModel):
    llm_id: str
    llm_name: str
    attempts: List[QuestionAttemptAnswer]

class QuestionResponse(BaseModel):
    id: str
    difficulty: float
    benchmark_category: str
    question: str
    llm_answers: List[QuestionLlmAnswer] 